### PR TITLE
refactor: add errorlint and refactor check error

### PR DESCRIPTION
**Pull request by hoank101**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved error message formatting across multiple modules to enhance error traceability using `%w` instead of `%v` or `%s`.
  - Updated error handling logic to use `errors.Is` for better error comparison.

### CODE REVIEW

**Comment by hoank101** [comment]
> heya @hoank101, we discussed and we agreed we'd like to try these linters out. Can't push to your branch so could you possibly either fix the conflicts or allow maintainers to push?

hi @DimitrisJim can you approve PR

**Comment by hoank101** [comment]
> heya @hoank101, we discussed and we agreed we'd like to try these linters out. Can't push to your branch so could you possibly either fix the conflicts or allow maintainers to push?

hi @DimitrisJim @crodriguezvega can you review PR?

File: modules/core/04-channel/keeper/upgrade.go [hunk 1]
```diff
@@ -980,7 +981,8 @@ func (k *Keeper) abortUpgrade(ctx sdk.Context, portID, channelID string, err err
 
 	// in the case of application callbacks, the error may not be an upgrade error.
 	// in this case we need to construct one in order to write the error receipt.
-	upgradeError, ok := err.(*types.UpgradeError)
+	var upgradeError *types.UpgradeError
```
[the remark this answers is not in the document]
Comment (gjermundgaraba) [reply modules/core/04-channel/keeper/upgrade.go:984]: Hmm, not sure, but right off the bat it feels like the old check is easier to read.

There is actually a helper method that might be better: `types.IsUpgradeError()`?

**Comment by crodriguezvega** [comment]
PR is stale, so closing for now.

**Pull request closed without merging**